x/poolmanager/router_test.go

```diff
@@ -662,51 +768,36 @@ func (suite *KeeperTestSuite) TestEstimateMultihopSwapExactAmountIn() {
 			poolmanagerKeeper := suite.App.PoolManagerKeeper
 			poolDefaultSwapFee := sdk.NewDecWithPrec(1, 2) // 1%
 
-			// Prepare 4 pools,
-			// Two pools for calculating `MultihopSwapExactAmountIn`
-			// and two pools for calculating `EstimateMultihopSwapExactAmountIn`
-			suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
-			suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
-
-			firstEstimatePoolId := suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
-			secondEstimatePoolId := suite.PrepareBalancerPoolWithPoolParams(balancer.PoolParams{
-				SwapFee: poolDefaultSwapFee,
-				ExitFee: sdk.NewDec(0),
-			})
+			setupPoolsStrategy := getSetupPoolsStrategy(suite, test.poolType, poolDefaultSwapFee)
+			firstEstimatePoolId, secondEstimatePoolId := setupPoolsStrategy()
```

Comment:
What is the reason for returning a callback function only to call it on the next line? Can we have only `setupPools` function that returns the ids?